### PR TITLE
Fix to prevent adding empty structs into materials.

### PR DESCRIPTION
**Pull request by jagathprakash**
Signed-off-by: jagathprakash <31057312+jagathprakash@users.noreply.github.com>

# Changes

Prior to this change if there was no CHAINS-GIT_URL or CHAINS-GIT_COMMIT results in PipelineRun object, we would add an empty struct to the materials section of the SLSA provenance.
This fixes this to prevent the addition of the empty structs.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [X] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
NONE
```

### CODE REVIEW

File: pkg/chains/formats/intotoite6/pipelinerun/provenance_test.go [hunk 1]
```diff
@@ -33,16 +33,18 @@ import (
...
 // Global pro is only read from, never modified
 var pro *objects.PipelineRunObject
+var pro_structured_results *objects.PipelineRunObject
 var e1BuildStart = time.Unix(1617011400, 0)
 var e1BuildFinished = time.Unix(1617011415, 0)
 
 func init() {
-	pro = createPro()
+	pro = createPro("../testdata/pipelinerun1.json")
+	pro_structured_results = createPro("../testdata/pipelinerun_structured_results.json")
```
Comment (wlynch) [inline pkg/chains/formats/intotoite6/pipelinerun/provenance_test.go:42]:
```suggestion
	proStructuredResults = createPro("../testdata/pipelinerun_structured_results.json")
```

File: pkg/chains/formats/intotoite6/pipelinerun/provenance_test.go [hunk 2]
```diff
@@ -33,16 +33,18 @@ import (
...
 }
 
-func createPro() *objects.PipelineRunObject {
+func createPro(prJson string) *objects.PipelineRunObject {
```
Comment (wlynch) [inline pkg/chains/formats/intotoite6/pipelinerun/provenance_test.go:45]:
```suggestion
func createPro(path string) *objects.PipelineRunObject {
```

**Review by wlynch — COMMENTED** [review]
LGTM! Just some minor nits.

**Comment by jagathprakash** [comment]
> LGTM! Just some minor nits.

Thanks for the review. Addressed all comments.

**Comment by wlynch** [comment]
Needs a push? 😅

**Comment by jagathprakash** [comment]
> Needs a push? 😅

yes 😬 

done 😃

**Review by wlynch — APPROVED** [review]
Thanks!

/lgtm

**Pull request merged**